### PR TITLE
Refined the mongo.js config file, Closes #6855

### LINKED ISSUE #6855: [OTHER] Fix mongo db initial connection config
### What would you like to share?

Currently, to establish a connection with the mongoDB instance, we use the `connectMongo` function in the `config/mongo.js` file. A new connection is not established when there is an existing connection. Please refer to the attached screenshot:
Screenshot 2023-05-13 at 11 28 14 PM

`hasConnection` variable is assigned in line 24 and reassigned in line 28. 

Proposal:

1. replace `hasConnection` with `connection` variable which stores the result of `await mongoose.connect(....` which returns an object with the current connection's information.
2. Remove the return and re assignment of `hasConnection` since we don't use the connection information anywhere.
3. Improvement: extract `LINKFREE_MONGO_CONNECTION_STRING` variable using `const { } = params.env` and re use it wherever needed. This way, we don't have to use `process.env.....` everywhere.
4. Remove the usage of `process.env.CA_CERT` as it is not being used anymore.
5. Fix `import logger from "../config/logger";` to `import logger from "./logger";` since the `logger.js` file exists in the same directory.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Chalithya**
## Fixes Issue

Closes #6855

## Changes proposed

1. replace hasConnection with connection variable which stores the result of await mongoose.connect(.... which returns an object with the current connection's information.
2.Remove the return and re assignment of hasConnection since we don't use the connection information anywhere.
3.Improvement: extract LINKFREE_MONGO_CONNECTION_STRING variable using const { } = params.env and re use it wherever needed. This way, we don't have to use process.env..... everywhere.
4.Remove the usage of process.env.CA_CERT as it is not being used anymore.
5.Fix import logger from "../config/logger"; to import logger from "./logger"; since the logger.js file exists in the same directory.

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Note to reviewers

I removed returns inside try catch as it returns a Promise and throwed an error inside the catch block to reduce the number of lines in the code. 

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6885"></a>

### CODE REVIEW

File: config/mongo.js [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
-    );
-    hasConnection = true;
-    logger.info("MongoDB connected");
-    return hasConnection;
-  } catch (e) {
-    hasConnection = false;
-    logger.error(e, "DB connection failed");
+  try {
+    const options = { autoIndex: true, family: 4 };
+    connection = await mongoose.connect(LINKFREE_MONGO_CONNECTION_STRING,options);
```
Comment (mnvsk97) [inline config/mongo.js:22]:
```suggestion
    connection = await mongoose.connect(LINKFREE_MONGO_CONNECTION_STRING, options);
```

File: config/mongo.js [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
+    logger.info("MongoDB connected:", connection.name);
```
Comment (mnvsk97) [inline config/mongo.js:23]: Hey 👋 we can also improve this log to something like 
`DB connection successful:` to keep it in sync with the error log in line 25.

File: config/mongo.js [hunk 3]
```diff
@@ -1,36 +1,29 @@
...
+
+  if (!LINKFREE_MONGO_CONNECTION_STRING) {
     throw new Error(
       "Please define the LINKFREE_MONGO_CONNECTION_STRING environment variable (if local add to .env file)"
     );
   }
 
-  if (hasConnection) {
-    return hasConnection;
+  if (connection) {
```
Comment (mnvsk97) [inline config/mongo.js:16]: This check to see if the connection exists should be at the start of the function. If a connection exists, the function can return from there itself.

**Comment by mnvsk97** [comment]
Hey [Chalithya](https://github.com/Chalithya) 👋 
I've added a few comments. Please have a look and let me know if you me to elaborate on any of the comments. Thanks for working on this issue !

File: config/mongo.js (hunk 3 above)
Comment (Chalithya) [reply config/mongo.js:16]: Yeah. It's true. We don't need to check connection strings if there is existing connection available

**Comment by eddiejaoude** [comment]
Great collaboration 💪

File: config/mongo.js [hunk 4]
```diff
@@ -1,36 +1,29 @@
 import mongoose from "mongoose";
-import * as fs from "fs";
+import { env } from "process";
```
Comment (eddiejaoude) [inline config/mongo.js:2]: I don't think this is needed

File: config/mongo.js [hunk 5]
```diff
@@ -1,36 +1,29 @@
...
-    );
-    hasConnection = true;
-    logger.info("MongoDB connected");
-    return hasConnection;
-  } catch (e) {
-    hasConnection = false;
-    logger.error(e, "DB connection failed");
+    const options = { autoIndex: true, family: 4 };
+    connection = await mongoose.connect(LINKFREE_MONGO_CONNECTION_STRING, options);
+    logger.info("DB connection successful:", connection.name);
```
Comment (eddiejaoude) [inline config/mongo.js:23]: I think this is missing a `return connection` otherwise it won't work on the first call when there is no connection

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good 👍 

Great collaboration! I left some small comments

**Comment by eddiejaoude** [comment]
I will merge and tweak in a temporary branch

**Pull request merged**